elide-async/src/main/java/com/yahoo/elide/async/models/AsyncAPI.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2020, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.async.models;
+
+import com.yahoo.elide.annotation.ComputedAttribute;
+import com.yahoo.elide.annotation.CreatePermission;
+import com.yahoo.elide.annotation.Exclude;
+import com.yahoo.elide.annotation.UpdatePermission;
+import com.yahoo.elide.async.service.AsyncAPIUpdateThread;
+import com.yahoo.elide.async.service.AsyncExecutorService;
+import com.yahoo.elide.security.User;
+
+import org.apache.http.NoHttpResponseException;
+
+import lombok.Data;
+
+import java.net.URISyntaxException;
+import java.util.Date;
+import java.util.UUID;
+
+import javax.persistence.Column;
+import javax.persistence.DiscriminatorColumn;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.Id;
+import javax.persistence.Inheritance;
+import javax.persistence.InheritanceType;
+import javax.persistence.PreUpdate;
+import javax.persistence.Transient;
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Pattern;
+
+/**
+ * Base Model Class for Async Query.
+ */
+@Entity
+@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
+@DiscriminatorColumn
+@Data
+public abstract class AsyncAPI implements PrincipalOwned {
```

Comment:
We could make this non-abstract and expose it also as an Elide entity.  Benefits include:
1. Being able to see all the queries in one place
2. We can consolidate cleaners and cancel threads because they can query the base class (instead of the subclass).